netbox/utilities/forms/rendering.py

```diff
@@ -0,0 +1,49 @@
+import random
+import string
+from functools import cached_property
+
+__all__ = (
+    'InlineFields',
+    'ObjectAttribute',
+    'TabbedFieldGroups',
+)
+
+
+class FieldGroup:
+
+    def __init__(self, label, *field_names):
+        self.field_names = field_names
+        self.label = label
+
+
+class InlineFields(FieldGroup):
+    pass
+
+
+class TabbedFieldGroups:
+
+    def __init__(self, *groups):
+        self.groups = [
+            FieldGroup(*group) for group in groups
+        ]
+
+        # Initialize a random ID for the group (for tab selection)
+        self.id = ''.join(
+            random.choice(string.ascii_lowercase + string.digits) for _ in range(8)
+        )
+
+    @cached_property
+    def tabs(self):
+        return [
+            {
+                'id': f'{self.id}_{i}',
+                'title': group.label,
+                'fields': group.field_names,
+            } for i, group in enumerate(self.groups, start=1)
+        ]
+
+
+class ObjectAttribute:
```

Comment:
I wonder if instead of ObjectAttribute you could use readonly_fields and just display those as values, seems more consistent with Django and django-admin usage.

Reply:
The `ObjectAttribute` class is needed to display values that are not tied to actual form fields, but exist as attributes on its related instance.